lib/kubernetes-deploy/kubernetes_resource/horizontal_pod_autoscaler.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+module KubernetesDeploy
+  class HorizontalPodAutoscaler < KubernetesResource
+    TIMEOUT = 5.minutes
+    RECOVERABLE_CONDITIONS = %w(ScalingDisabled FailedGet)
+
+    def deploy_succeeded?
+      scaling_active_condition["status"] == "True"
+    end
+
+    def deploy_failed?
+      return false unless exists?
+      recoverable = RECOVERABLE_CONDITIONS.any? { |c| scaling_active_condition.fetch("reason", "").start_with?(c) }
+      able_to_scale_condition["status"] == "False" || (scaling_active_condition["status"] == "False" && !recoverable)
```

Comment:
I don't think the "able to scale" condition is relevant at all (same in `status`). It's talking about whether or not the target in question can be scaled at this particular moment (based on things like whether or not it was _just_ scaled a moment ago), not on the general validity of the autoscaling configuration. I see you have a test that implies that "able to scale - false; FailedGet" means the target resource may not exist, but couldn't failing on that cause a race condition, since the HPA doesn't get deployed after deployments? Couldn't it also mean the request for that resource (transiently) failed, i.e. should be excluded on the same grounds as other FailedGetXs? Is "Scaling active" actually true when the target doesn't exist?

Reply:
I actually think `FailedGetScale` is relevant since when we are failing with that reason there isn't an  'AbleToScale' condition to look at. I'll scope this down to just that condition